aws/sandboxes/heka/output/kinesis.lua

```diff
@@ -0,0 +1,236 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Kinesis Producer Output
+
+## Sample Configuration
+```lua
+filename            = "kinesis_single.lua"
+message_matcher     = "TRUE"
+ticker_interval     = 0 -- set to 1 to timeout/send any packed messages
+
+streamName              = "foobar"
+-- partitionField       = "Uuid"
+-- credentialProvider   = "INSTANCE"
+-- clientConfig         = nil
+-- roleArn              = nil
+
+-- Due to the way Kinesis is billed it makes senses to pack multiple items
+-- into a single message to reduce the PUT Payload Unit charges. Warning: if
+-- pack is greater than zero and and pack_partitions is true (multiple batches)
+-- the check pointing system cannot track them individualy i.e. in case of a
+-- failure items could be lost.
+pack_percentage = 0 -- fill percentage of the put request (0 (no packing) - <100).
+         -- The record won't be sent unless the current PUT Payload unit is this
+         -- percentage full or the record has reached its maximum size or the
+         -- timed out is triggered.
+pack_delimiter = nil    -- Used to add a record delimiter, if necessary
+pack_partitions = false -- create a batch for each partition
+
+-- Specify a module that will encode/convert the Heka message into its output representation.
+encoder_module = "encoders.heka.protobuf" -- default
+```
+--]]
+
+require "aws.kinesis"
+require "string"
+require "table"
+require "xxhash"
+
+local PUT_MAX   = 1024 * 1024
+local PUT_UNIT  = 25 * 1024
+
+local streamName            = read_config("streamName") or error"streamName must be set"
+local partitionField        = read_config("partitionField") or "Uuid"
```

Comment:
is Uuid a good default to use if pack_partitions is true but the partition field isn't set? Should we just error out if pack partitions is enabled and the field isn't specified?

Reply:
Sounds good, updating